### PR TITLE
Reset headers

### PR DESCRIPTION
**Pull request by vincentchalamon**
Required by https://github.com/Behatch/contexts/pull/164
See https://github.com/Behatch/contexts/issues/163

### CODE REVIEW

**Comment by Shine-neko** [comment]
ping @fabpot

File: Goutte/Client.php [hunk 1]
```diff
@@ -62,6 +62,11 @@ public function removeHeader($name)
         unset($this->headers[$name]);
     }
 
+    public function resetHeaders()
```
Comment (fabpot) [inline Goutte/Client.php]: I would prefer to have a `reset()` method that resets all settings, including auth for instance.

**Comment by webaaz** [comment]
Need merge for Behatch/contexts#164

Thanks

**Comment by jubianchi** [comment]
This PR seems good. I see only one thing we could change to keep everything consistent: the `Client` already has a `resetAuth` method to let users reset only auth settings. Why not do the same for headers?

I think the proposed `resetHeaders` method is useful if we only want to reset headers.

Then, the `reset` method would just have to call both `reset*` methods:

```php
public function reset()
{
    $this
        ->resetAuth()
        ->resetHeaders();
}
```

This will also keep things simple if reseting headers ever involves a much complex logic.

BTW, it's ok for me, unfortunately @webaaz I don't have the right to merge here and I would like to have @fabpot's go before doing so.

**Comment by vincentchalamon** [comment]
@jubianchi Fixed, thanks for your feedback

File: Goutte/Client.php [hunk 2]
```diff
@@ -112,7 +125,7 @@ protected function doRequest($request)
                 $requestOptions['body'] = $content;
             } else {
                 if ($files = $request->getFiles()) {
-                    $requestOptions['multipart'] = [];
+                    $requestOptions['multipart'] = array();
```
Comment (jubianchi) [inline Goutte/Client.php]: there is no need to revert bakc to the long array syntax. This package [requires PHP `>=5.5.0`](https://github.com/FriendsOfPHP/Goutte/blob/master/composer.json#L15)

File: Goutte/Client.php [hunk 3]
```diff
@@ -62,6 +62,19 @@ public function removeHeader($name)
...
 
+    public function resetHeaders()
+    {
+        $this->headers = array();
+
+        return $this;
+    }
+
+    public function reset()
+    {
```
Comment (stof) [inline Goutte/Client.php]: you also need to call the parent reset method, otherwise cookies are not reset properly

File: Goutte/Client.php [hunk 4]
```diff
@@ -157,9 +170,9 @@ protected function addPostFiles(array $files, array &$multipart, $arrayName = ''
                 $name = $arrayName.'['.$name.']';
             }
 
-            $file = [
+            $file = array(
```
Comment (stof) [inline Goutte/Client.php]: why changing this ?

Comment (vincentchalamon) [inline Goutte/Client.php]: Because of fabbot.io

File: Goutte/Client.php (hunk 3 above)
Comment (vincentchalamon) [inline Goutte/Client.php]: There is no `reset` method on parent class. Did you mean `restart` method?

File: Goutte/Client.php (hunk 2 above)
Comment (vincentchalamon) [inline Goutte/Client.php]: It's required by fabbot.io

File: Goutte/Client.php (hunk 3 above)
Comment (stof) [inline Goutte/Client.php]: hmm, indeed, I confused names

Comment (vincentchalamon) [inline Goutte/Client.php]: So better rename `reset` method name in `restart` to be consistent?

File: Goutte/Client.php (hunk 4 above)
Comment (fabpot) [inline Goutte/Client.php]: @stof is right, these changes should be reverted. Sorry about the confusion, fabbot is quite dumb sometimes :)

**Comment by webaaz** [comment]
@vincentchalamon You know why fabbot.io refuse the check?

**Comment by vincentchalamon** [comment]
@webaaz Check this comment from @fabpot: https://github.com/FriendsOfPHP/Goutte/pull/261#discussion_r87832202

**Comment by fabpot** [comment]
Thank you @vincentchalamon.

**Comment by webaaz** [comment]
thank for the merge!

**Pull request closed without merging**